File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(num)
    @num = num
    @num_array = (1..@num).to_a
  end

  def square_of_sum
    sum = 0
    @num_array.each { |num| sum += num}
    sum**2
  end

  def sum_of_squares
    squares = []
    sum = 0
    @num_array.each { |num| squares << num**2 }
    squares.each { |num| sum += num}
    sum
  end

  def difference
    self.square_of_sum - self.sum_of_squares
  end
end

module BookKeeping
 VERSION=3
end

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(count):
    count_sum = sum([x for x in range(1, count + 1)])
    return count_sum**2

def sum_of_squares(count):
    return sum([x**2 for x in range(1, count + 1)])

def difference(count):
    return square_of_sum(count) - sum_of_squares(count)
